teuthology/task/install/rpm.py

```diff
@@ -62,6 +64,45 @@ def _remove(ctx, config, remote, rpm):
     else:
         remote.run(args='sudo yum clean expire-cache')
 
+def _yum_addrepo(remote, repo_list):
+    """
+    Add yum repos to the remote system.
+
+    :param remote: remote node where to add packages
+    :param repo_list: list of dictionaries with keys 'name', 'url', etc.
+    :return:
+    """
+    for repo in repo_list:
```

Comment:
See `task/internal/redhat.py` for an example, but you can download/write the file locally, copy it to the remote, and then "sudo cp" it to its destination. It will look cleaner than this. Some of the code from `ShamanProject` can also probably be refactored / reused to handle pulling the repo file down.

Reply:
@dillaman why this is so complicated?

Reply:
In general the idea was to use commands like `zypper` or `yum`.

Reply:
The end goal is to be able to pull a dynamic repo using the shaman API (i.e. pull the latest master branch repo for "ceph-iscsi" for the EL8 environment).

Reply:
@dillaman I don't understand how is this code related for shaman? The shaman repos are used in different place. This code is used when you provide test repo list in suite yaml or in override block.

Reply:
@kshtsk it might not directly related to shaman. but if the only use case is to use the repos exposed by shaman RESTful service. it might be a lot simpler to reuse `ShamanProject` , as we can query shaman for a ready-to-use repo file, instead of composing a `dnf` command.

Reply:
from the description in the tracket ticket (https://tracker.ceph.com/issues/47431):

> For the ceph-iscsi test suite, we need to add extra ceph-iscsi and tcmu-runner repos from shaman to install the packages.

i conclude that these repos are exposed by shaman. i think the code is confusing. as if we want to packages exposed by shaman, instead of looking for a "repo", what this method should accept `project`, `branch`, `sha1`, `dist`, `codename`. as shaman expects them in, for instance, following RESTful API:

> https://shaman.ceph.com/api/repos/${project}/${branch}/${sha1}/${dist}/${codename}/repo

Reply:
@tchaikov 

The link from `https://shaman.ceph.com/api/repos/XXX` won't work for the `dnf config-manager --add-repo https://shaman.ceph.com/api/repos/XXX.../repo`.

The `dnf config-manager` is trying to find an `example.repo` file from the URL, if there is no it will just create a repo and set the `baseurl` to the URL specified and then it will be like:

```
name=created by dnf config-manager from https://1.chacra.ceph.com/repos/XXX/repo
baseurl=https://shaman.ceph.com/api/repos/XXX/repo
enabled=1
```  

But the URL here couldn't find the packages we needed.

Maybe we should do:
such as for the repo in `https://1.chacra.ceph.com/repos/tcmu-runner/master/ebffeff02ac35ef3f09371c41038f5c83e993ad1/centos/8/flavors/default/repo`
```
    - install:
        repos:
          -  prefix: "https://1.chacra.ceph.com/r/"
             project: "tcmu-runner"
             branch: "mater"
             sha1: "XXXXX"
             dist: "centos"
             relase: "8"
             flavers: "default"
             archs: "x86_64, noarch, RPMS"
```

Then do `sudo dnf config-manager --add-repo ${url}/${project}/${branch}/${sha1}/${disto}/${release}/flavors/${flaver}/${arch}

Reply:
> The link from https://shaman.ceph.com/api/repos/XXX won't work for the dnf config-manager --add-repo https://shaman.ceph.com/api/repos/XXX.../repo.

i am not suggesting use RESTful API along with `dnf config-manager`. i think you missed my comment at https://github.com/ceph/teuthology/pull/1558/files#r490040959

Reply:
I understand what the ticket is about but `repos` is opposite what you want, of course it can be used by the code which is use Shaman class, but the feature of the repos is exactly to override what is used by Shaman, for example, we downstream just use it with obs and by external scheduler when we override the repos.

Reply:
> Then do `sudo dnf config-manager --add-repo ${url}/${project}/${branch}/${sha1}/${disto}/${release}/flavors/${flaver}/${arch}

instead of this, we can add a feature for `url` and `name` to render job properties, like:
```
install:
  repos:
    - name: "tcmu-runner-{branch}"
      url: "https://1.chacra.ceph.com/r/tcmu-runner/{branch}/{sha1}/{os_type}/{os_version}/flavors/{flavor}/{arch}
```